Angus/build updates

veggiesaurus opened the pull request:
Closes #321 and closes #706 

This is a rather large change to the build, because it now uses `create-react-app@3.4.3`, which comes with new build scripts. Some other changes:

- allows us to import SCSS files directly, no need to run all the CSS-watching stuff
- uses eslint (with typescript plugins) instead of tslint
- uses browserlist to ensure that we compile to a sensible target, rather than supporting ancient browsers

There are a few stricter eslint rules by default, and this has required a number of minor changes to the code. Hopefully this doesn't break anything. 

@kswang1029 can you do some testing of the frontend in general?
@ajm-asiaa can you try build electron apps using this branch?
@duidae @panchyo0 @TienHao can you check through the changes to the code itself? Most were suggested directly by eslint.

note: this seems to break the breadcrumbs fix by @ajm-asiaa, I'll look into solving this during the review process

duidae committed:
Jason/issue 435 (#1221)

* Check whether cursor is inside image or not

* Correct with rounding

veggiesaurus committed:
WIP fixing eslint config

veggiesaurus committed:
linting fixes

veggiesaurus committed:
reverted readme change

veggiesaurus committed:
fixed git-info typo

veggiesaurus committed:
info update

ajm-ska commented:
@veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
<img width="572" alt="Screenshot 2020-10-07 at 20 29 43" src="https://user-images.githubusercontent.com/20802551/95331059-04b91d80-08dc-11eb-9651-9c9c3ab113de.png">

veggiesaurus commented:
> @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> <img width="572" alt="Screenshot 2020-10-07 at 20 29 43" src="https://user-images.githubusercontent.com/20802551/95331059-04b91d80-08dc-11eb-9651-9c9c3ab113de.png">
> 

Perhaps you need to force a hard reload?

ajm-ska commented:
> > @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> > <img alt="Screenshot 2020-10-07 at 20 29 43" width="572" src="https://user-images.githubusercontent.com/20802551/95331059-04b91d80-08dc-11eb-9651-9c9c3ab113de.png">
> 
> Perhaps you need to force a hard reload?

That doesn't seem to work (holding shift and hitting reload). 
I see the automatic builds on the [Mix 'N Match](http://carta.asiaa.sinica.edu.tw/builds/) page have the same issue.

veggiesaurus committed:
reverted change to homepage setting for package.json

veggiesaurus commented:
> > > @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> > > <img alt="Screenshot 2020-10-07 at 20 29 43" width="572" src="https://user-images.githubusercontent.com/20802551/95331059-04b91d80-08dc-11eb-9651-9c9c3ab113de.png">
> > 
> > 
> > Perhaps you need to force a hard reload?
> 
> That doesn't seem to work (holding shift and hitting reload).
> I see the automatic builds on the [Mix 'N Match](http://carta.asiaa.sinica.edu.tw/builds/) page have the same issue.

It might be because I removed the `homepage` setting in `package.json`. I've added it back, can you see if that solves it?

ajm-ska commented:
> > > > @veggiesaurus Does the result of `npm run build` work for you? I only seem to get a blank screen. The debug console shows the following:
> > > > <img alt="Screenshot 2020-10-07 at 20 29 43" width="572" src="https://user-images.githubusercontent.com/20802551/95331059-04b91d80-08dc-11eb-9651-9c9c3ab113de.png">
> > > 
> > > 
> > > Perhaps you need to force a hard reload?
> > 
> > 
> > That doesn't seem to work (holding shift and hitting reload).
> > I see the automatic builds on the [Mix 'N Match](http://carta.asiaa.sinica.edu.tw/builds/) page have the same issue.
> 
> It might be because I removed the `homepage` setting in `package.json`. I've added it back, can you see if that solves it?

Yes. That fixed the problem! It works now. It appears to work when put into Electron too.

ajm-ska reviewed (approved):
The building aspects seem to be fine. The production build works now and the Electron app version can still be produced. Using the Docker build process seems to work fine too.

kswang1029 commented:
I am trying to explore the GUI from time to time to see if there are odd things and report here.
PART1
1. the checkbox of the molecular line transition table in the spectral line query widget and preference-log event is broken.
![Screen Shot 2020-10-08 at 8 48 49 AM](https://user-images.githubusercontent.com/20819712/95403008-9f007c00-0943-11eb-9fc4-b3fa7903764f.png)
![Screen Shot 2020-10-08 at 9 17 25 AM](https://user-images.githubusercontent.com/20819712/95404274-2ac7d780-0947-11eb-9e92-6f77e50e6caa.png)

2. breadcrumbs in the file browser is broken
![Screen Shot 2020-10-08 at 9 03 40 AM](https://user-images.githubusercontent.com/20819712/95403584-36b29a00-0945-11eb-8373-0738d9959ad7.png)

duidae commented on src/components/Shared/ProfilerInfo/ProfilerInfoComponent.tsx:
```diff
@@ -1,19 +1,23 @@
-    props.info?.forEach((info, index) => {
-        infoString = index === 0 ? info : (infoString + ", " + info);
-    });
+    let infoTags = null;
```
seems previous PR #1216  is not merged

duidae commented on src/stores/FrameStore.ts:
```diff
@@ -1072,16 +1073,10 @@ export class FrameStore {
             }
         }
 
-        const imageX = Math.round(cursorPosImageSpace.x);
```
PR #1221 is also missing

duidae reviewed (changes requested):
PR #1216 & #1221 are missing

panchyo0 commented on package.json:
```diff
@@ -40,22 +42,23 @@
         "react-resize-detector": "^4.2.1",
         "react-rnd": "^9.2.0",
+        "react-scripts": "3.4.3",
         "react-scripts-ts": "^2.17.0",
```
should we remove "react-scripts-ts" from package.json?

veggiesaurus committed:
removed react-scripts-ts

veggiesaurus committed:
cherry-pick of PR #1216

veggiesaurus committed:
removed unnecessary package import

veggiesaurus committed:
import CSS files instead of SCSS to solve breadcrumbs and checkbox issue

veggiesaurus committed:
removed crumbs fix for prod build

veggiesaurus commented:
The breadcrumbs and checkbox issues have been fixed now. And we no longer need the `crumbs_fix` workaround :+1: 

@kswang1029 can you give it another try when you get the chance?

veggiesaurus committed:
removed react-google-login dependency and reformatted json

kswang1029 commented:
@veggiesaurus confirmed with mix n match: both breadcrumbs and checkbox work properly.
